### ISSUE TITLE
final commands not being run in behaviour space before results output to file

### ISSUE THREAD

**Issue by j2bryson** [NetLogo/NetLogo#924]
Hi, 

There's a simple hack around this so there's no urgency, but I had the below in the "final commands" part of behaviour space & it got ignored, so I'm moving it to "measure runs with these reporters", since anyway I was only running that at the end of the runs (thank you so much yet again for adding that tick box!)

> set avg-trust avg-trust / (ticks - start-average)
> set avg-rate avg-rate / (ticks - start-average)
> set avg-demand avg-demand / (ticks - start-average)
> set avg-priv-owned avg-priv-owned / (ticks - start-average)

**Comment by SethTisue** [comment]
how do you know the commands were ignored? I doubt there's a bug in BehaviorSpace causing final commands to be completely ignored. perhaps the commands ran, but a different time than you expected? perhaps http://ccl.northwestern.edu/netlogo/docs/behaviorspace.html should clarify the timing?

**Comment by j2bryson** [comment]
I know they were being ignored because I was keeping a sum that was
intended as an average, and the "finally" should have divided it by the
number of ticks, so it was pretty obvious that it was the number of ticks x
larger than I expected :-)  The problem was that the program stopped
running because a count exceeded the arithmetic / numerical limits of
netlogo, so maybe I should be happy that BS kept going at all, but as I
said I had kind of a java set of expectations about either having my
variables not reported at all or captured & divided.

Anyway, I've figured out a way to limit the size of the variables so it is
all working at the moment (and I made the CogSci deadline :-)

Joanna


On 1 February 2016 at 15:59, Seth Tisue <notifications@github.com> wrote:

> how do you know the commands were ignored? I doubt there's a bug in
> BehaviorSpace causing final commands to be completely ignored. perhaps the
> commands ran, but a different time than you expected? perhaps
> http://ccl.northwestern.edu/netlogo/docs/behaviorspace.html should
> clarify the timing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NetLogo/NetLogo/issues/924#issuecomment-178186426>.
>



-- 
J J Bryson
http://www.cs.bath.ac.uk/~jjb/

**Comment by ERSUCC** [comment]
I don't think this is an issue anymore. I ran a test experiment where I tried to do various large arithmetic operations, and each time it displayed an error dialog reading "math operation produced a number too large for NetLogo". This is also the behavior in the Command Center.

**Comment by TheBizzle** [comment]
I tried making a BehaviorSpace experiment that had really nothing to it, except running (nothing) for one tick, and then performing the following "final commands":

```netlogo
let big-boy 9e307
show "Hello"
show big-boy + big-boy           ; Returns Infinity
show big-boy + big-boy + big-boy ; Uses Infinity as an operand, and then returns Infinity 
show 1 / (big-boy - big-boy)     ; Generates division by zero error, as expected
```

But, even in NetLogo 5.3.0 (which came out just before this bug report), everything seemed fine.  In more-recent versions of NetLogo, we throw  an error instead of returning `Infinity` as a numeric value.  But, other than that... the behavior seems to be the same, and I'm unable to find any version of NetLogo that replicates the behavior described here.

I suppose I'll close this as "can't reproduce", until a reproducible example comes in.
